### PR TITLE
fix(agoric-cli): handle another vstorage error variation

### LINKED ISSUE #9033: error code 18 ... pruned from inter bid list
## Describe the bug

`agops inter bid list` crashes

Similar symptoms have come up in discussion with Oracle operators.

## To Reproduce
1. `AGORIC_NET=emerynet inter bid list --from agoric1ldmtatp24qlllgxmrsjzcpe20fvlkp448zcuce`
2. See error

```console
$ AGORIC_NET=emerynet inter bid list --from agoric1ldmtatp24qlllgxmrsjzcpe20fvlkp448zcuce
(Error#1)
Error#1: error code 18 reading data of published.wallet.agoric1ldmtatp24qlllgxmrsjzcpe20fvlkp448zcuce: failed to load state at height 4067923; version mismatch on immutable IAVL tree; version does not exist. Version has either been pruned, or is for a future block height (latest height: 4110876): invalid request

  at packages/agoric-cli/src/lib/rpc.js:83:23
  at async Object.readAt (packages/agoric-cli/src/lib/rpc.js:122:19)
  at async Object.readFully (packages/agoric-cli/src/lib/rpc.js:144:38)
  at async storedWalletState (packages/agoric-cli/src/lib/wallet.js:228:21)
  at async Promise.all (index 1)
  at async Command.<anonymous> (packages/agoric-cli/src/commands/inter.js:566:34)
```

## Expected behavior

List bids from that account.

## Platform Environment
 - what OS are you using? what version of Node.js?

```console
$ uname -a
Linux bldbox 5.15.0-92-generic #102-Ubuntu SMP Wed Jan 10 09:33:48 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux
$ node --version
v18.17.1
```

 - is there anything special/unusual about your platform?
 - what version of the Agoric-SDK are you using? (run `git describe --tags --always`)

@agoric/cosmic-swingset@0.41.3-1359-gfd0382de8 fd0382de8

## Additional context

[Listing your open bids: inter bid list](https://github.com/Agoric/agoric-sdk/blob/master/packages/inter-protocol/test/auction/README-bidding-cli.md#listing-your-open-bids-inter-bid-list) in [Inter Protocol Liquidation Bidding Test Tool](https://github.com/Agoric/agoric-sdk/blob/master/packages/inter-protocol/test/auction/README-bidding-cli.md#listing-your-open-bids-inter-bid-list)

cc @otoole-brendan @michaelfig

### PR DESCRIPTION
**Pull request by dckc**
fixes #9033

avoid:

```
Error#1: error code 18 reading data of published.wallet.agoric1l...: failed to load state at height 14007...; version mismatch on immutable IAVL tree; version does not exist. Version has either been pruned, or is for a future block height (latest height: 14026...): invalid request

  at packages/agoric-cli/src/lib/rpc.js:83:23
  at async Object.readAt (packages/agoric-cli/src/lib/rpc.js:122:19)
  at async Object.readFully (packages/agoric-cli/src/lib/rpc.js:144:38)
  at async storedWalletState (packages/agoric-cli/src/lib/wallet.js:228:21)
  at async Promise.all (index 1)
  at async
  Command.<anonymous> (packages/agoric-cli/src/commands/inter.js:566:34)
```

## Description

When exploring the history of data at a vstorage key, `err.codespace === 'sdk' && err.code === 38` is sort of the first-class signal for "there is no more" (_according to what? I'm not sure_). But we don't seem to always get that sort of error. We have also accepted any error whose message matches `/unknown request/`. With this change, we also take `/pruned/` as such a signal.

DRAFT until:

 - [ ] is there a spec for these errors that we could / should cite in this code?
 - [ ] given the cost of a test, is it OK to land this without one?
 - [ ] to fully address #9033, is there something on the chain side that we should fix? or is this enough? That is: should I change "closes" to "refs"?

### Security / Scaling Considerations

n / a

### Documentation Considerations

If this fix is to-spec, then there are none. Otherwise, perhaps some disclaimer is in order, at least as a code comment.

### Testing Considerations

A fix normally comes with a test. A unit test would be nice. But this testing CLI code isn't factored to facilitate unit testing. It kinda needs a whole running blockchain. So a test is probably more trouble than it's worth for a ~1 line fix.

### Upgrade Considerations

The change that introduced the symptoms of #9033 was evidently missing some upgrade considerations.

### CODE REVIEW

**Comment by dckc** [comment]
hm... maybe I forgot to `yarn build` or something?

```
$ AGORIC_NET=main inter bid list --from agoric1l...
node:internal/errors:496
    ErrorCaptureStackTrace(err);
    ^

Error [ERR_MODULE_NOT_FOUND]: Cannot find module '/Users/connolly/projects/agoric-sdk/node_modules/@agoric/cosmic-proto/dist/codegen/index.js' imported from /Users/connolly/projects/agoric-sdk/packages/agoric-cli/src/lib/chain.js
    at new NodeError (node:internal/errors:405:5)
    at finalizeResolution (node:internal/modules/esm/resolve:327:11)
    at moduleResolve (node:internal/modules/esm/resolve:946:10)
    at defaultResolve (node:internal/modules/esm/resolve:1132:11)
    at nextResolve (node:internal/modules/esm/loader:163:28)
    at ESMLoader.resolve (node:internal/modules/esm/loader:835:30)
    at ESMLoader.getModuleJob (node:internal/modules/esm/loader:424:18)
    at ModuleWrap.<anonymous> (node:internal/modules/esm/module_job:77:40)
    at link (node:internal/modules/esm/module_job:76:36) {
  code: 'ERR_MODULE_NOT_FOUND'
}
```

**Comment by otoole-brendan** [comment]
@aj-agoric let's figure out a plan to expedite getting this deployed if it doesn't need a chain-halting upgrade. Users can't see their bids in CLI until this fix is deployed

**Comment by dckc** [comment]
@gibson042 I just added some questions to the PR description. Would you let me/us know what you think about them, please?

File: packages/agoric-cli/src/lib/rpc.js [hunk 1]
```diff
@@ -153,7 +153,8 @@ export const makeVStorage = (powers, config = networkConfig) => {
             // CosmosSDK ErrUnknownRequest; misrepresentation of the same until
             // https://github.com/Agoric/agoric-sdk/commit/dafc7c1708977aaa55e245dc09a73859cf1df192
             // TODO remove after upgrade-12
-            err.message.match(/unknown request/)
+            err.message.match(/unknown request/) ||
+            err.message.match(/pruned/)
```
Comment (gibson042) [inline packages/agoric-cli/src/lib/rpc.js]: Ugh, I was really hopeful about **abandoning** error message scans. In this case, the "_failed to load state at height_" text comes from [`func (app *BaseApp) createQueryContext`](https://github.com/agoric-labs/cosmos-sdk/blob/76e151914572fbefb0a6c7ebef1b51d464a9fcbe/baseapp/abci.go#L687-L694), which wraps the "_version mismatch on immutable IAVL tree; version does not exist. Version has either been pruned, or is for a future block height_" text from [`func (st *Store) GetImmutable`](https://github.com/agoric-labs/cosmos-sdk/blob/76e151914572fbefb0a6c7ebef1b51d464a9fcbe/store/iavl/store.go#L114) via [`func (rs *Store) CacheMultiStoreWithVersion`](https://github.com/agoric-labs/cosmos-sdk/blob/76e151914572fbefb0a6c7ebef1b51d464a9fcbe/store/rootmulti/store.go#L497) and unfortunately associates it with an "invalid request" error code. I have opened https://github.com/cosmos/cosmos-sdk/issues/19992 for that issue, and in the meantime I guess we want a corresponding update here.
```suggestion
            // CosmosSDK ErrInvalidRequest with particular message text;
            // misrepresentation of pruned data
            // TODO replace after incorporating a fix to
            // https://github.com/cosmos/cosmos-sdk/issues/19992
            (err.codespace === 'sdk' && err.code === 18 && err.message.match(/pruned/))
```

**Review by gibson042 — CHANGES_REQUESTED** [review]

**Review by gibson042 — APPROVED** [review]

**Comment by dckc** [comment]
glitch, I hope:

```
lerna ERR! E503 one of the uplinks is down, refuse to publish
```
-- [ci log #30969](https://github.com/Agoric/agoric-sdk/actions/runs/8787948814/job/24114351101?pr=9024)

**Pull request merged**